homeassistant/components/pglab/entity.py

```diff
@@ -0,0 +1,67 @@
+"""Entity for PG LAB Electronics."""
+
+from __future__ import annotations
+
+from pypglab.device import Device as PyPGLabDevice
+from pypglab.entity import Entity as PyPGLabEntity
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .discovery import PGLabDiscovery
+
+
+class PGLabEntity(Entity):
+    """Representation of a PGLab entity in Home Assistant."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        discovery: PGLabDiscovery,
+        device: PyPGLabDevice,
+        entity: PyPGLabEntity,
+    ) -> None:
+        """Initialize the class."""
+
+        self._id = entity.id
+        self._device_id = device.id
+        self._entity = entity
+        self._discovery = discovery
+
+        # Set the state update.
+        self._entity.add_state_update(self.state_updated)
```

Comment:
This is not OK, it means an entity which is disabled by user will still have active MQTT subscriptions and inject itself into the state machine.

The solution is to subscribe to entity specific topics and register the state update callback in `async_added_to_hass`, not here.

Please add tests which test:
a) We don't subscribe to topics which are needed only by a specific entity until needed (I'm not sure about the pglab topic schema, maybe this is not relevant)
b) There are no state changes for disabled entities

Since you've based this integration on tasmota, it may be helpful to look at tests in `tests/components/tasmota/test_sensor.py` which check behavior of disabled sensors

Reply:
I'm keeping self._entity.add_state_update(self.state_updated) as it is. It is NOT used to subscribe to  MQTT. It's save a reference to callback, used  for status change notification. When an entity is remove from HA (async_will_remove_from_hass), the state update callback is not called.  The fix to your previous comment (MQTT subscribe/unsubscribe symmetry) should make the core more readable.

Anyway, I'll add tests as you suggested.